File: app/src/main/java/com/example/dell/chargetimer/controls/StepNumberPicker.java
package com.example.dell.chargetimer.controls;

import android.app.Activity;
import android.support.annotation.IdRes;
import android.widget.NumberPicker;

import com.example.dell.chargetimer.helpers.Convert;

import java.util.List;

public class StepNumberPicker
{
    private final NumberPicker picker;
    private String[] values;

    public StepNumberPicker(Activity activity, @IdRes int viewId) {
        this((NumberPicker)activity.findViewById(viewId));
    }
    private StepNumberPicker(NumberPicker picker) {
        this.picker = picker;
    }

    public void setValues(List<String> values){
        setValues(Convert.toArray(values));
    }

    public void setValues(String[] values){
        this.values = values;
        picker.setDisplayedValues(null);
        picker.setMinValue(0);
        picker.setMaxValue(values.length - 1);
        picker.setDisplayedValues(values);
    }

    public void setOnValueChangedListener(NumberPicker.OnValueChangeListener listener){
        picker.setOnValueChangedListener(listener);
    }

    public String getValue(){
        if (values == null)
            throw new IllegalArgumentException("Allowed values not set");

        int index = picker.getValue();
        return values[index];
    }
    public void setValue(String value){
        if (values == null)
            throw new IllegalArgumentException("Allowed values not set");

        int index = getValueIndex(value);
        if (index < 0)
            throw new IllegalArgumentException("Passed value " + value + " out of range");

        picker.setValue(index);
    }

    private int getValueIndex(String s) {
        for (int i = 0; i < values.length; i++)
        {
            String item = values[i];
            if (item.equals(s))
                return i;
        }
        return -1;
    }
}

File: EvChargeTimer/app/src/main/java/com/chebuso/chargetimer/controls/StepNumberPicker.kt
package com.chebuso.chargetimer.controls

import android.app.Activity
import android.view.View
import android.widget.NumberPicker
import android.widget.NumberPicker.OnValueChangeListener
import androidx.annotation.IdRes

class StepNumberPicker private constructor(private val picker: NumberPicker) {
    private var values: Array<String?>? = null

    constructor(
        activity: Activity,
        @IdRes viewId: Int,
        values: List<String?>,
        selectedValue: String
    ) : this(activity.findViewById<View>(viewId) as NumberPicker){
        setValues(values.toTypedArray())
        value = selectedValue
    }

    fun setValues(values: Array<String?>) {
        this.values = values
        picker.setDisplayedValues(null)
        picker.setMinValue(0)
        picker.setMaxValue(values.size - 1)
        picker.setDisplayedValues(values)
    }

    fun setOnValueChangedListener(listener: OnValueChangeListener?) {
        picker.setOnValueChangedListener(listener)
    }

    var value: String
        get() {
            requireNotNull(values) { "Allowed values not set" }
            val index = picker.value
            val result = values!![index]
            return result!!
        }
        set(value) {
            requireNotNull(values) { "Allowed values not set" }
            val index = getValueIndex(value)
            require(index >= 0) { "Passed value $value out of range" }
            picker.value = index
        }

    private fun getValueIndex(s: String): Int {
        for (i in values!!.indices) {
            val item: String? = values!![i]
            if (item != null && item == s) return i
        }
        return -1
    }
}
